Katana/Store.swift

```diff
@@ -191,18 +191,31 @@ open class Store<S: State, D: SideEffectDependencyContainer>: PartialStore<S> {
   /// Whether the store is ready to execute operations
   public private(set) var isReady: Bool
   
-  /**
-   The dependencies used in the side effects
-   
-   - seeAlso: `SideEffect`
-   */
+  /// The dependencies used in the side effects
+  ///
+  ///- seeAlso: `SideEffect`
   public var dependencies: D!
   
+  /// The dependencies used to initialize katana
+  public struct Configuration {
```

Comment:
- Move this away from the location where variables are declared
- I think that `mainAsyncProvider` is weird. In particular, you want to say why you are asking for this "main async" value and you are not grabbing it directly.